adi/cn0566.py

```diff
@@ -0,0 +1,620 @@
+    ):
+        """ Set up devices, properties, helper methods, etc. """
+        if verbose is True:
+            print("attempting to open ADF4159, uri: ", str(uri))
+        adf4159.__init__(self, uri)
+        if verbose is True:
+            print("attempting to open ADAR1000 array, uri: ", str(uri))
+        sleep(0.5)
+        adar1000_array.__init__(
+            self, uri, chip_ids, device_map, element_map, device_element_map
+        )
+
+        if verbose is True:
+            print("attempting to open gpios , uri: ", str(uri))
+        sleep(0.5)
+        self.gpios = adi.one_bit_adc_dac(uri)
+
+        if verbose is True:
+            print("attempting to open AD7291 v/t monitor, uri: ", str(uri))
+        sleep(0.5)
+        self.monitor = adi.ad7291(uri)
+
+        """ Initialize all the class variables for the project. """
+        self.num_elements = 8  # eq. to 4 * len(list(self.devices.values()))
+        self.device_mode = "rx"  # Default to RX.
+        self.SignalFreq = 10492000000  # Frequency of source
+        self.Averages = 16  # Number of Avg to be taken.
+        self.phase_step_size = (
+            2.8125  # it is 360/2**number of bits. (number of bits = 6)
+        )
+        self.steer_res = (
+            2.8125  # It is steering resolution. This would be user selected value
+        )
+        self.c = 299792458  # speed of light in m/s
+        self.element_spacing = (
+            0.015  # element to element spacing of the antenna in meters
+        )
+        self.res_bits = 1  # res_bits and bits are two different var. It can be variable, but it is hardset to 1 for now
+        self.pcal = [
+            0.0 for i in range(0, (self.num_elements))
+        ]  # default phase cal value i.e 0
+        self.ccal = [
+            0.0,
+            0.0,
+        ]  # Gain compensation for the two RX channels in dB. Includes all errors, including the SDR's
+        self.gcal = [
+            1.0 for i in range(0, self.num_elements)
+        ]  # Per-element gain compensation, AFTER above channel compensation. Use to scale value sent to ADAR1000.
+        self.ph_deltas = [
+            0 for i in range(0, (self.num_elements) - 1)
+        ]  # Phase delta between elements
+        self.sdr = sdr  # rx_device/sdr that rx and plots
+        self.gain_cal = False  # gain/phase calibration status flag it goes True when performing calibration
+        self.phase_cal = False
+        # Scaling factors for voltage AD7291 monitor, straight from schematic.
+        self.v0_vdd1v8_scale = 1.0 + (10.0 / 10.0)  # Resistances in k ohms.
+        self.v1_vdd3v0_scale = 1.0 + (10.0 / 10.0)
+        self.v2_vdd3v3_scale = 1.0 + (10.0 / 10.0)
+        self.v3_vdd4v5_scale = 1.0 + (30.1 / 10.0)
+        self.v4_vdd_amp_scale = 1.0 + (69.8 / 10.0)
+        self.v5_vinput_scale = 1.0 + (30.1 / 10.0)
+        self.v6_imon_scale = 1.0  # LTC4217 IMON = 50uA/A * 20k = 1 V / A
+        self.v7_vtune_scale = 1.0 + (69.8 / 10.0)
```

Comment:
All public properties need docstrings. If they are not expected to be used by an end-user they should start with an underscore. If they are constants they ideally should also be in the class definition

Reply:
Done.